examples/ui/display_and_visibility.rs

```diff
@@ -0,0 +1,407 @@
+                                            },
+                                            background_color: BackgroundColor(palette[3]),
+                                            ..Default::default()
+                                        },))
+                                        .with_children(|parent| {
+                                            spawn_buttons(parent, target_ids.pop().unwrap());
+
+                                            parent.spawn(NodeBundle {
+                                                style: Style {
+                                                    size: Size::all(Val::Px(100.)),
+                                                    ..Default::default()
+                                                },
+                                                ..Default::default()
+                                            });
+                                        });
+                                });
+                        });
+                });
+        });
+}
+
+fn spawn_button<T>(parent: &mut ChildBuilder, text_style: TextStyle, target: Entity)
+where
+    T: Default + std::fmt::Debug + Send + Sync + 'static,
+    Target<T>: TargetUpdate,
+{
+    parent.spawn((
+        ButtonBundle {
+            style: Style {
+                size: Size::height(Val::Px(24.)),
+                align_self: AlignSelf::FlexStart,
+                ..Default::default()
+            },
+            background_color: BackgroundColor(Color::BLACK.with_a(0.5)),
+            ..Default::default()
+        },
+        Target::<T>::new(target),
+        Text::from_section(
+            format!("{}::{:?}", Target::<T>::NAME, T::default()),
+            text_style,
+        )
+        .with_alignment(TextAlignment::Center),
+        CalculatedSize::default(),
+    ));
+}
+
+fn buttons_handler<T>(
+    mut left_panel_query: Query<&mut <Target<T> as TargetUpdate>::TargetComponent>,
+    mut visibility_button_query: Query<(&mut Text, &Target<T>, &Interaction), Changed<Interaction>>,
+) where
+    T: Send + Sync,
+    Target<T>: TargetUpdate + Component,
+{
+    for (mut text, target, interaction) in visibility_button_query.iter_mut() {
+        if matches!(interaction, Interaction::Clicked) {
+            let mut target_value = left_panel_query.get_mut(target.id).unwrap();
+            text.sections[0].value = target.update_target(target_value.as_mut());
+        }
+    }
+}
+
+fn text_hover(
+    mut text_query: Query<(&mut Text, &mut BackgroundColor, &Interaction), Changed<Interaction>>,
+) {
+    for (mut text, mut background_color, interaction) in text_query.iter_mut() {
+        match interaction {
+            Interaction::Hovered => {
+                text.sections[0].style.color = Color::YELLOW;
+                *background_color = BackgroundColor(Color::BLACK.with_a(0.6));
+            }
+            _ => {
+                text.sections[0].style.color = Color::WHITE;
```

Comment:
**Nit:** just an idea, it would be nice to show the text in "red" when the value is `None`/`Hidden`. I've tried with `Color::RED/ORANGE_RED` but the contrast is awful. This seems to work better (the contrast with the orange background is still bad) `Color::rgb(1.0, 0.7, 0.7)`. It can be improved by changing the background colors, maybe you could just use a gradient of blues?

![image](https://user-images.githubusercontent.com/188612/223526187-0a5ab9cd-6434-4228-9f87-ca32cb5a1fc3.png)

I've tested it like this… which is quite hacky 😅 

```rust
text.sections[0].style.color = if text.sections[0].value.contains("None")
    || text.sections[0].value.contains("Hidden")
{
    Color::rgb(1.0, 0.7, 0.7)
} else {
    Color::WHITE
};
```

Reply:
Yes Red seems better, I'll change it.

Reply:
Changed the palette as well, the gold panel was a bit odd.